tests/ui-msrv/transmute-ref-dst-unsized.stderr

```diff
@@ -11,6 +25,10 @@ note: required by a bound in `DST_UNSIZED::transmute`
 17 | const DST_UNSIZED: &[u8] = transmute_ref!(&[0u8; 1]);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `DST_UNSIZED::transmute`
    = note: this error originates in the macro `transmute_ref` (in Nightly builds, run with -Z macro-backtrace for more info)
+help: consider relaxing the implicit `Sized` restriction
+   |
+211|             const fn transmute<'u, 't: 'u, T: 't, U: 'u + ?Sized>(t: &'t T) -> &'u U {
+   |                                                         ++++++++
```

Comment:
I *think* this is only occurring because `transmute` is defined in the expansion. Could we make this go away by defining it in `macro_util` instead?